pkg/resources/grant_ownership_acceptance_test.go

```diff
@@ -579,6 +556,245 @@ func TestAcc_GrantOwnership_InvalidConfiguration_MultipleTargets(t *testing.T) {
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "account_role_name", accountRoleName),
+					resource.TestCheckResourceAttr(resourceName, "on.0.object_type", "MATERIALIZED VIEW"),
+					resource.TestCheckResourceAttr(resourceName, "on.0.object_name", materializedViewFullyQualifiedName),
+					resource.TestCheckResourceAttr(resourceName, "id", fmt.Sprintf("ToAccountRole|%s||OnObject|MATERIALIZED VIEW|%s", accountRoleFullyQualifiedName, materializedViewFullyQualifiedName)),
+					checkResourceOwnershipIsGranted(&sdk.ShowGrantOptions{
+						To: &sdk.ShowGrantsTo{
+							Role: sdk.NewAccountObjectIdentifier(accountRoleName),
+						},
+					}, sdk.ObjectTypeMaterializedView, accountRoleName, fmt.Sprintf("%s.%s.%s", databaseName, schemaName, materializedViewName)),
+				),
+			},
+			{
+				ConfigDirectory:   acc.ConfigurationDirectory("TestAcc_GrantOwnership/OnObject_MaterializedView_ToAccountRole"),
+				ConfigVariables:   configVariables,
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAcc_GrantOwnership_OnTasks(t *testing.T) {
+	t.Skip("will be unskipped in the following grant ownership prs")
+	// 		- Tasks on bulk
+}
+
+func TestAcc_GrantOwnership_OnAllTasks(t *testing.T) {
+	t.Skip("will be unskipped in the following grant ownership prs")
+	// 		- Tasks on bulk
+}
+
+func TestAcc_GrantOwnership_RoleBasedAccessControlUseCase(t *testing.T) {
+	accountRoleName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	databaseName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	schemaName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	userName := getCurrentUser(t)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			// We have to make it in two steps, because provider blocks cannot contain depends_on meta-argument
+			// that are needed to grant the role to the current user before it can be used.
+			// Additionally, only the Config field can specify a configuration with custom provider blocks.
+			{
+				Config: roleBasedAccessControlUseCaseConfig(accountRoleName, databaseName, userName, schemaName, false),
+			},
+			{
+				Config: roleBasedAccessControlUseCaseConfig(accountRoleName, databaseName, userName, schemaName, true),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PostApplyPostRefresh: []plancheck.PlanCheck{
+						plancheck.ExpectEmptyPlan(),
+					},
+				},
+			},
+		},
+	})
+}
+
+func roleBasedAccessControlUseCaseConfig(accountRoleName string, databaseName string, userName string, schemaName string, withSecondaryProvider bool) string {
```

Comment:
Hmm, I agree that splitting those functions is better, but in this case, I think I would leave it as is. It's only used in one test case and splitting wouldn't be that beneficial, because the `b()` config is strongly connected to the `a()` one, so It wouldn't make much sense to use them separately.

Reply:
I disagree, this will cause more similar non-reusable functions to arise and it could be split in to a few reusable ones.

Helpers rework is incoming, though, so we can leave this as is for now. Resolve after reading.